internal/handlers/api.go

```diff
@@ -1,18 +1,34 @@
 package handlers
 
 import (
+	"net/http"
+
 	chimiddle "github.com/go-chi/chi/middleware"
 	"github.com/go-chi/chi/v5"
 	"github.com/thirdweb-dev/indexer/internal/middleware"
 )
 
 func Handler(r *chi.Mux) {
 	r.Use(chimiddle.StripSlashes)
-
+	r.Use(middleware.Authorization)
 	r.Route("/api", func(router chi.Router) {
-		router.Use(middleware.Authorization)
-		router.Route("/v1", func(r chi.Router) {
-			r.Get("/blocks", GetBlocks)
-		})
+		// might consolidate all variants to one handler function
+		// Wild card queries
+		router.Get("/transactions/{chainId}", GetTransactions)
+		router.Get("/events/{chainId}", GetLogs)
+
+		// contract scoped queries
+		router.Get("/transactions/{chainId}/{contractAddress}", GetTransactionsWithContract)
+		router.Get("/events/{chainId}/{contractAddress}", GetEventsWithContract)
+
+		// signature scoped queries
+		router.Get("/transactions/{chainId}/{contractAddress}/{functionSig}", GetTransactionsWithContractAndSignature)
+		router.Get("/events/{chainId}/{contractAddress}/{functionSig}", GetEventsWithContractAndSignature)
```

Comment:
Doesn't it make more sense from a REST viewpoint to have `chainId` as the first resource?
e.g. `api/{chainId}/transactions`

Reply:
done